imagekitio/url.py

```diff
@@ -130,6 +130,7 @@ def get_signature_timestamp(expiry_seconds: int = None) -> int:
 
     @staticmethod
     def get_signature(private_key, url, url_endpoint, expiry_timestamp: int) -> str:
+        url = Url.encode_string_if_required(url)
```

Comment:
Wouldn't it be better to call `Url.encode_string_if_required(url)` this down the line when on the `replaced_url` which we actually used for signature generation? It would save the unnecessary logic in `custom_encodeURIComponent` of replacing `url_endpoint` from `url`, thereby simplifying the code.

Reply:
addressed